accsaber-business/src/main/java/de/ixsen/accsaber/business/RankedMapService.java

```diff
@@ -93,6 +93,27 @@ public void addNewRankedMap(String beatSaverId, Long leaderBoardId, String diffi
         this.scoreRepository.saveAll(nowRankedScores);
     }
 
+    public void removeRankedMap(Long leaderboardId) {
+        Optional<RankedMap> map = this.rankedMapRepository.findById(leaderboardId);
+        if(map.isEmpty()){
+            throw new AccsaberOperationException(ExceptionType.RANKED_MAP_NOT_FOUND,
+                    "The ranked map"  + leaderboardId + " does not currently exist.");
+        }
+
+        Song song = map.get().getSong();
+        List<RankedMap> rankedMaps = song.getRankedMaps();
+        rankedMaps.remove(map);
+        song.setRankedMaps(rankedMaps);
```

Comment:
Yout don't need to set the ranked maps for the song here. You could just do:
song.getRankedMaps().remove(map.get());